components/flow-orchestration-framework/org.wso2.carbon.identity.flow.execution.engine/src/main/java/org/wso2/carbon/identity/flow/execution/engine/FlowExecutionService.java

```diff
@@ -90,12 +95,27 @@ public FlowExecutionStep executeFlow(String tenantDomain, String applicationId,
                 }
             }
             if (STATUS_COMPLETE.equals(step.getFlowStatus())) {
+
+                if (REGISTRATION_FLOW_TYPE.equals(context.getFlowType())) {
+                    Map<String, String> userClaims =
+                            context.getFlowUser() != null ? context.getFlowUser().getClaims() : null;
+                    FrameworkUtils.publishEventOnUserRegistrationSuccess(userClaims, tenantDomain);
+                }
+
                 FlowExecutionEngineUtils.removeFlowContextFromCache(flowId);
             } else {
                 FlowExecutionEngineUtils.addFlowContextToCache(context);
             }
             return step;
         } catch (FlowEngineException e) {
+
+            if (context != null && REGISTRATION_FLOW_TYPE.equals(context.getFlowType())) {
```

Comment:
Since claims can be null, should we publish the event only within the `if` block?